crates/router/src/core/payment_methods/cards.rs

```diff
@@ -45,7 +51,26 @@ pub async fn create_payment_method(
     customer_id: &str,
     payment_method_id: &str,
     merchant_id: &str,
+    connector: Option<&api::ConnectorData>,
+    token: Option<String>,
 ) -> errors::CustomResult<storage::PaymentMethod, errors::StorageError> {
+    let pm_metadata = token
+        .map(|token| -> errors::CustomResult<_, errors::StorageError> {
+            let metadata = payment_methods::PaymentMethodMetadata {
+                connector: connector
```

Comment:
This is a plain structure and can be accommodate only one connector and token right?

Reply:
While deserializing, i have deserialized to the vector of this struct, so that I will get the token for that particular connector easily.So plain structure would suffice here.